sdks/python/apache_beam/internal/pickler.py

```diff
@@ -68,10 +68,12 @@ def load_session(file_path):
 def set_library(selected_library=DEFAULT_PICKLE_LIB):
   """ Sets pickle library that will be used. """
   global desired_pickle_lib
-  if selected_library == USE_DILL and desired_pickle_lib != dill_pickler:
+  if (selected_library == USE_DILL) != (desired_pickle_lib == dill_pickler):
```

Comment:
Did you mean "(selected_library == USE_DILL) **and**"

Logic is bit hard to follow.

Reply:
If the new value of (pickler is dill) is different than the old value, we have to set (or unset) the hooks. Added a comment to clarify.